File: cmd/client/src/api/booking.ts
const API_BASE = '/api'

export interface CreateBookingRequest {
  groupId: string
  planId: string
  participantName: string
  participantPhone: string
  participantEmail: string
}

export interface Booking {
  id: string
  groupId: string
  planId: string
  participantName: string
  participantPhone: string
  participantEmail: string
  bookingDate: string
  createdAt: string
}

export interface Group {
  id: string
  day: string
  time: string
  startTime: string
  endTime: string
  maxParticipants: number
  currentParticipants: number
  instructor?: string
}


export async function createBooking(data: CreateBookingRequest): Promise<Booking> {
  const res = await fetch(`${API_BASE}/bookings`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  })

  if (!res.ok) {
    const text = await res.text()
    throw new Error(text || `Request failed: ${res.status}`)
  }

  return res.json()
}

export async function getGroups(): Promise<Group[]> {
  const res = await fetch(`${API_BASE}/groups`)

  if (!res.ok) {
    throw new Error(`Failed to fetch groups: ${res.status}`)
  }

  return res.json()
}

File: internal/server/models/booking.go
package models

import "time"

type Booking struct {
	ID               string    `json:"id" db:"id"`
	GroupID          string    `json:"groupId" db:"group_id"`
	PlanID           string    `json:"planId" db:"plan_id"`
	ParticipantName  string    `json:"participantName" db:"participant_name"`
	ParticipantPhone string    `json:"participantPhone" db:"participant_phone"`
	ParticipantEmail string    `json:"participantEmail" db:"participant_email"`
	BookingDate      time.Time `json:"bookingDate" db:"booking_date"`
	CreatedAt        time.Time `json:"createdAt" db:"created_at"`
}

type CreateBookingRequest struct {
	GroupID          string `json:"groupId"`
	PlanID           string `json:"planId"`
	ParticipantName  string `json:"participantName"`
	ParticipantPhone string `json:"participantPhone"`
	ParticipantEmail string `json:"participantEmail"`
}
